services/api/config/arvados_config.rb

```diff
@@ -278,7 +278,7 @@ def arrayToHash cfg, k, v
 # For config migration, we've previously populated the PostgreSQL
 # section of the config from database.yml
 #
-database_url = CGI.escape("postgresql://#{$arvados_config["PostgreSQL"]["Connection"]["user"]}:"+
+database_url = "postgresql://#{CGI.escape $arvados_config["PostgreSQL"]["Connection"]["user"]}:"+
```

Comment:
This ought to solve your issue but since we're at it would probably be worth adding CGI.escape to `password` and `dbname` just to be on the safe side.

Reply:
Ok let me add that.